src/main/java/htsjdk/samtools/SamReader.java

```diff
@@ -381,7 +381,7 @@ public PrimitiveSamReaderToSamReaderAdapter(final PrimitiveSamReader p, final Sa
             this.resource = resource;
         }
 
-        PrimitiveSamReader underlyingReader() {
+        public PrimitiveSamReader underlyingReader() {
```

Comment:
Can you explain why you can't just use the existing `iterator(final SAMFileSpan chunks)` on the `PrimitiveSamReaderToSamReaderAdapter`, instead of making `underlyingReader()` public, calling it on the adapter, and then calling your new `getIterator(long firstRecordPointer)` method on it.